simple_history/models.py

```diff
@@ -233,13 +252,64 @@ def copy_fields(self, model):
             fields[field.name] = field
         return fields
 
+    @property
+    def _history_change_reason_field(self):
+        if self.history_change_reason_field:
+            # User specific field from init
+            history_change_reason_field = self.history_change_reason_field
+        elif getattr(
+            settings, "SIMPLE_HISTORY_HISTORY_CHANGE_REASON_USE_TEXT_FIELD", False
+        ):
+            # Use text field with no max length, not enforced by DB anyways
+            history_change_reason_field = models.TextField(null=True)
+        else:
+            # Current default, with max length
+            history_change_reason_field = models.CharField(max_length=100, null=True)
+
+        return history_change_reason_field
+
+    @property
+    def _history_id_field(self):
```

Comment:
Hmm why are these two property methods needed in this PR? Doesn't seem related to the user changes.

Reply:
Ah it looks to be general refactoring from below. Definitely into pulling the logic into their own methods, but I'm not as big a fan of `@property`. Unless there's a compelling reason to do so, my preference would be to invoke them as regular methods.

Reply:
Was trying to mimic how I’ve seen similar things done in Django source code. Don’t really have a preference though, so would be happy to change these into ‘get’ methods